internal/discord/handlers.go

```diff
@@ -0,0 +1,251 @@
+			Name:        "reason",
+			Description: "Reason for ban",
+			Required:    false,
+		},
+	},
+}
+
+var UnbanCommand = &discordgo.ApplicationCommand{
+	Name:                     "unban",
+	DefaultMemberPermissions: &banPermission,
+	Description:              "Unban the specified user.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionUser,
+			Name:        "user",
+			Description: "User being unbanned",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "reason",
+			Description: "Reason for unban",
+			Required:    false,
+		},
+	},
+}
+
+var RemoveNicknameCommand = &discordgo.ApplicationCommand{
+	Name:                     "removenickname",
+	DefaultMemberPermissions: &mutePermission,
+	Description:              "Remove the nickname of the specified user.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionUser,
+			Name:        "user",
+			Description: "User whose nickname to rename",
+			Required:    true,
+		},
+	},
+}
+
+var SetNicknameCommand = &discordgo.ApplicationCommand{
+	Name:                     "setnickname",
+	DefaultMemberPermissions: &mutePermission,
+	Description:              "Change the nickname of the specified user.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionUser,
+			Name:        "user",
+			Description: "User whose nickname to rename",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "nickname",
+			Description: "Nickname to rename user to",
+			Required:    true,
+		},
+	},
+}
+
+var SlowmodeCommand = &discordgo.ApplicationCommand{
+	Name:                     "slowmode",
+	DefaultMemberPermissions: &channelPermission,
+	Description:              "Add slowmode to current channel.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionInteger,
+			Name:        "duration",
+			Description: "Slowmode interval duration in seconds",
+			Required:    true,
+		},
+	},
+}
+
+var SlowmodeOffCommand = &discordgo.ApplicationCommand{
+	Name:                     "slowmodeoff",
+	DefaultMemberPermissions: &channelPermission,
+	Description:              "Remove slowmode from current channel.",
+}
+
+var PurgeCommand = &discordgo.ApplicationCommand{
+	Name:                     "purge",
+	DefaultMemberPermissions: &messagePermission,
+	Description:              "Delete a number of messages from a channel.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionChannel,
+			Name:        "channel",
+			Description: "Channel to purge",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionInteger,
+			Name:        "message-number",
+			Description: "Number of messages to purge",
```

Comment:
we might need to implement a max for this, im not sure how many resources are requred for a normal purge

Reply:
resources shouldn't be an issue, it would take longer to purge more messages